codalab/worker/worker.py

```diff
@@ -622,6 +626,16 @@ def initialize_run(self, bundle, resources):
                 bundle=bundle,
                 bundle_path=os.path.realpath(bundle_path),
                 bundle_dir_wait_num_tries=Worker.BUNDLE_DIR_WAIT_NUM_TRIES,
+                bundle_profile_stats={
+                    RunStage.PREPARING: self.init_stage_stats(),
+                    RunStage.RUNNING: self.init_stage_stats(),
+                    RunStage.CLEANING_UP: self.init_stage_stats(),
+                    RunStage.UPLOADING_RESULTS: self.init_stage_stats(),
+                    RunStage.FINALIZING: self.init_stage_stats(),
+                    # note: this is not sent over the api, but
+                    # the start time needs to be set
+                    RunStage.FINISHED: self.init_stage_stats(),
```

Comment:
I think it would be cleaner to remove RunStage.FINISHED altogether from this dictionary, and then in the line where we call `self.start_stage_stats(uuid, self.runs[uuid].stage)`, add an if statement to ensure it's called only if `self.runs[uuid].stage != RunState.FINISHED`. This way, we're only storing what we need in this dictionary.

Reply:
We would add this check only when starting stage statistics, correct?